config/configChecker.go

```diff
@@ -0,0 +1,86 @@
+package config
+
+import (
+	"fmt"
+
+	logger "github.com/multiversx/mx-chain-logger-go"
+)
+
+var log = logger.GetOrCreate("configChecker")
+
+func SanityCheckEnableEpochsStakingV4(cfg *Configs) error {
+	enableEpochsCfg := cfg.EpochConfig.EnableEpochs
+	err := checkStakingV4EpochsOrder(enableEpochsCfg)
+	if err != nil {
+		return err
+	}
+
+	numOfShards := cfg.GeneralConfig.GeneralSettings.GenesisMaxNumberOfShards
+	return checkStakingV4MaxNodesChangeCfg(enableEpochsCfg, numOfShards)
+}
+
+func checkStakingV4EpochsOrder(enableEpochsCfg EnableEpochs) error {
+	stakingV4StepsInOrder := (enableEpochsCfg.StakingV4Step1EnableEpoch < enableEpochsCfg.StakingV4Step2EnableEpoch) &&
+		(enableEpochsCfg.StakingV4Step2EnableEpoch < enableEpochsCfg.StakingV4Step3EnableEpoch)
+
+	if !stakingV4StepsInOrder {
+		return errStakingV4StepsNotInOrder
+	}
+
+	stakingV4StepsInExpectedOrder := (enableEpochsCfg.StakingV4Step1EnableEpoch == enableEpochsCfg.StakingV4Step2EnableEpoch-1) &&
+		(enableEpochsCfg.StakingV4Step2EnableEpoch == enableEpochsCfg.StakingV4Step3EnableEpoch-1)
+	if !stakingV4StepsInExpectedOrder {
+		log.Warn("staking v4 enable epoch steps should be in cardinal order " +
```

Comment:
Please return error. Why do you want something like this for playground ? You can have scripts which change to code here in case of playground, it is used in other places.

Reply:
yes let's return the error here as we won't be testing something other than will eventually be on the mainnet. Use fmt.Errorf here and on the other 3 locations to easily pinpoint the error and have a quick fix on the configs

Reply:
I was not sure if to consider these "soft errors" or not, so I've decided to let them as they are and fix it after review if necessary.
Thanks for suggestion, I've returned errors for all cases